Implementation of group creation function

kawa18sima opened the pull request:
# WHAT
Implementation of the function to create a group to chat.
Processing related after group creation is also implemented.
# WHY
It's necessary to chat in the group.

Commit:
Add routing of groups

Changed from only to except along with update, create of groups

Because root access was not possible, change

Commit:
Addition of links of users edit and groups new to index of messages

messagesのindexのアイコンに
groups new
users edit
のリンクの追加

Commit:
user_groups model generation

Use the rails g model user_groups command

Commit:
Create user_groups table

User_id column
Group_id column
Create

Commit:
Add Model Association

By adding user_groups model
Add Association

Commit:
Add description of clearfix

Add clearfix description to groups.scss
Add linefeed at the last line of clearfix

Commit:
Edit "controller" and "view of new" of groups

Add processing for group creation

Commit:
Edit left side view of messages index

Change to show the group to which you belong

kawa18sima commented:
ここでユーザの追加機能も作りました。
お願いします！！
<img width="1440" alt="2017-05-03 23 53 36" src="https://cloud.githubusercontent.com/assets/24731933/25666862/bc536984-305c-11e7-91f9-83586a2db831.png">
<img width="1440" alt="2017-05-03 23 57 16" src="https://cloud.githubusercontent.com/assets/24731933/25666865/bf355cca-305c-11e7-8bc5-7aa22f03c45f.png">
<img width="657" alt="2017-05-03 23 56 46" src="https://cloud.githubusercontent.com/assets/24731933/25666868/c1b90b36-305c-11e7-9d4d-26dcfae427ea.png">
<img width="755" alt="2017-05-03 23 57 06" src="https://cloud.githubusercontent.com/assets/24731933/25666870/c3170136-305c-11e7-8b21-812d65dfc445.png">

kinshotomoya commented on app/views/groups/new.html.haml:
```diff
@@ -1,29 +1,22 @@
           #chat-group-user-22.chat-group-user.clearfix
-            %input{name: "chat_group[user_ids][]", type: "hidden", value: "22"}/
-            %p.chat-group-user__name seo_kyohei
+            =f.collection_check_boxes :user_ids, User.all, :id, :name
```
view側でUse.allすると、SQLがもう一回走って、処理が遅くなってしまうので
controller側で定義しましょう！

kinshotomoya replied:
あと、ログインしているユーザーの欄は、最初からチェックされているようにしましょう！

kinshotomoya commented on app/views/messages/index.html.haml:
```diff
@@ -7,15 +7,11 @@
-      %li.group__list__name
-        %h2 aeaaaaaaa
-        %p  まだメッセージがありません
+      - @groups.each do |group|
```
部分テンプレート、render使おう！

kinshotomoya reviewed (commented):
コメントしました！
ちゃんと保存できていますね！
あとは、細かいところを修正していきましょう！

Commit:
Edit "controller" and "view of new" of groups

Change "User.all" part of collection_check_boxes to execution by
controller
The user who is logged in is changed from the beginning to the checked
state

Commit:
Template the left side of messages index

Replace the left side of the message's index with a partial template
named left-group-list.
Change display of user name to "current_user.name" display

kawa18sima commented:
指摘された部分編集しました。
お願いします！！

kinshotomoya commented on app/controllers/groups_controller.rb:
```diff
@@ -2,6 +2,7 @@ class GroupsController < ApplicationController
 
   def new
     @group = Group.new
+    @groups = User.all
```
@users にしよう！！！！

kinshotomoya commented on app/views/messages/_left-group-list.html.haml:
```diff
@@ -0,0 +1,14 @@
+    = fa_icon "pencil-square-o"
+.group
+  %ul.group__list
+    - groups.each do |group|
```
renderでcollection使おう！！！！
eachの役割をしてくれる！

kinshotomoya reviewed (commented):
コメントしました！
下記修正してください！

Commit:
Fix groups instance variable name

I made it by mistake once
@groups -> @users

Commit:
Modify messages index

Change the description of each statement to a description using reder
collection

kawa18sima commented:
編集しました！！
凡ミスに気をつけます。

kinshotomoya commented:
見ました！！
一点だけ！
render collection定義している場所がおかしいように見えます！
部分テンプレートの中でrender collectionを使わないです！！！！！！！！

Commit:
Editing the partial template of messages

I was calling a partial template in a partial template, but I changed
part of it to index and made only a part of each as a partial template

kawa18sima commented:
編集しました！
render collectionの部分だけをテンプレート化して、それ以外はmessagesのindexに戻しました。

kinshotomoya commented on app/views/messages/index.html.haml:
```diff
@@ -1,5 +1,14 @@
+      = fa_icon "pencil-square-o"
+  .group
+    %ul.group__list
+      = render partial: "group-lists",collection: @groups,as: :group
```
おっけです！！！！！！！！
,のあとにスペースないのがきになるので、半角スペースあけといてください！

kinshotomoya reviewed (commented):
見ました！
下記修正して、マージで！！！！！！！！！！！！！！！！！！！！！！！！！！！！！！
頑張りましょウ！！！！！！！

Commit:
Leave space after ","

I left space.
